packages/rrweb/src/record/observers/canvas/canvas-manager.ts

```diff
@@ -167,6 +182,89 @@ export class CanvasManager implements CanvasManagerInterface {
       unblockSelector,
       true,
     );
+    const rafId = this.takeSnapshot(
+      false,
+      fps,
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      options.dataURLOptions,
+    );
+
+    this.resetObservers = () => {
+      canvasContextReset();
+      cancelAnimationFrame(rafId);
+    };
+  }
+
+  private initCanvasMutationObserver(
+    win: IWindow,
+    blockClass: blockClass,
+    blockSelector: string | null,
+    unblockSelector: string | null,
+  ): void {
+    this.startRAFTimestamping();
+    this.startPendingCanvasMutationFlusher();
+
+    const canvasContextReset = initCanvasContextObserver(
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      false,
+    );
+    const canvas2DReset = initCanvas2DMutationObserver(
+      this.processMutation.bind(this),
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+    );
+
+    const canvasWebGL1and2Reset = initCanvasWebGLMutationObserver(
+      this.processMutation.bind(this),
+      win,
+      blockClass,
+      blockSelector,
+      unblockSelector,
+      this.mirror,
+    );
+
+    this.resetObservers = () => {
+      canvasContextReset();
+      canvas2DReset();
+      canvasWebGL1and2Reset();
+    };
+  }
+
+  private manualSnapshot(canvasElement?: HTMLCanvasElement) {
+    const rafId = this.takeSnapshot(
+      true,
+      this.options.sampling === 'all' ? 2 : this.options.sampling || 2,
+      this.options.win,
+      this.options.blockClass,
+      this.options.blockSelector,
+      this.options.unblockSelector,
+      this.options.dataURLOptions,
```

Comment:
I'd be curious of the build size differences between this and if we were to set an intermediate var for `this.options`. I think the intermediate var would then be able to be minified/mangled

```
const {options} = this;
```

Reply:
the build size decreased a little bit!